access/msteams/plugindata.go

```diff
@@ -15,42 +18,58 @@ type PluginData struct {
 
 // TeamsMessage represents sent message information
 type TeamsMessage struct {
-	ID          string
-	Timestamp   string
-	RecipientID string
+	ID          string `json:"id"`
+	Timestamp   string `json:"ts"`
+	RecipientID string `json:"rid"`
 }
 
 // DecodePluginData deserializes a string map to PluginData struct.
-func DecodePluginData(dataMap map[string]string) PluginData {
+func DecodePluginData(dataMap map[string]string) (PluginData, error) {
 	data := PluginData{}
+	var errors []error
 
 	data.AccessRequestData = plugindata.DecodeAccessRequestData(dataMap)
 
-	if channelID, timestamp := dataMap["channel_id"], dataMap["timestamp"]; channelID != "" && timestamp != "" {
-		data.TeamsData = append(data.TeamsData, TeamsMessage{ID: channelID, Timestamp: timestamp})
-	}
 	if str := dataMap["messages"]; str != "" {
 		for _, encodedMsg := range strings.Split(str, ",") {
-			parts := strings.Split(encodedMsg, "/")
-			if len(parts) == 3 {
-				data.TeamsData = append(data.TeamsData, TeamsMessage{ID: parts[0], Timestamp: parts[1], RecipientID: parts[2]})
+			decodedMsg, err := base64.StdEncoding.DecodeString(encodedMsg)
+			if err != nil {
+				// Backward compatibility
+				// TODO(hugoShaka): remove in v12
+				parts := strings.Split(encodedMsg, "/")
+				if len(parts) == 3 {
+					data.TeamsData = append(data.TeamsData, TeamsMessage{ID: parts[0], Timestamp: parts[1], RecipientID: parts[2]})
+				}
+			} else {
+				msg := &TeamsMessage{}
+				err = json.Unmarshal(decodedMsg, msg)
+				if err != nil {
+					errors = append(errors, err)
+				}
+				data.TeamsData = append(data.TeamsData, *msg)
 			}
```

Comment:
:nail_care: 
Swap the `else` with a `continue` in the `if` block
This reduces the indentation level which helps readability (a subjective term :sweat_smile: )

```suggestion
				continue
			}
			
			msg := &TeamsMessage{}
			err = json.Unmarshal(decodedMsg, msg)
			if err != nil {
				errors = append(errors, err)
			}
			data.TeamsData = append(data.TeamsData, *msg)
```